### PR TITLE
[MPS] Fix in-place add and sub with alpha == 0.0

### LINKED ISSUE #96065: [MPS] add_ and sub_ drop gradients when alpha=0 for >=1.12.1
### 🐛 Describe the bug

Both `torch.Tensor.add_` and `torch.Tensor.sub_` drop gradients when alpha is set to 0. In the example below, only the last assertion fails.

```
import torch

def test_grad(alpha, device):
    other = torch.tensor([1.], device=device)
    input = torch.tensor([1.], device=device, requires_grad=True)

    with torch.no_grad():
        input.add_(other, alpha=alpha)

    assert input.requires_grad

test_grad(alpha=1, device="cpu")
test_grad(alpha=0, device="cpu")
test_grad(alpha=1, device="mps")
test_grad(alpha=0, device="mps") # assertion fails with >=1.12.1
```

This does not occur for 1.12.0, it only becomes an issue with 1.12.1, 1.13.1, and nightly (2.0.0.dev20230301).

### Versions

PyTorch version: 1.13.1
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 13.2.1 (arm64)
GCC version: Could not collect
Clang version: 15.0.7
CMake version: Could not collect
Libc version: N/A

Python version: 3.10.6 (main, Aug 29 2022, 10:06:59) [Clang 13.1.6 (clang-1316.0.21.2.5)] (64-bit runtime)
Python platform: macOS-13.2.1-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Apple M1 Max

Versions of relevant libraries:
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.24.2
[pip3] torch==1.13.1
[pip3] torchtext==0.13.1
[conda] Could not collect

cc @kulinseth @albanD @malfet @DenisVieriu97 @razarmehr @abhudev

### PR DESCRIPTION
**Pull request by qqaatw**
Apart from fixing the below issue, this PR integrates the test for `sub` into the test for `add` as they are implemented using the same template.

Fixes #96065

### CODE REVIEW

**Review by kulinseth — APPROVED** [review]
Looks good.
